frontend/src/components/WofButton.vue

```diff
@@ -0,0 +1,89 @@
+<template>
+  <button :class="`button button--${type}`" @click="onClick()">
+    <div v-if="this.$slots.icon" :class="this.$slots.default ? 'button--icon' : ''"><slot name="icon"/></div>
```

Comment:
Let's use first approach - no slot for icon. Icon should be in button displayed conditionally.
Add prop for 'size' which will set size for text and for icon.
Also, if you want set icon in WofIcon, don't use "style". Use "icon" prop (see WofIcon code)

Reply:
But then I would have to specify its class and I don't want to do that, right?

Reply:
Class?

Reply:
I am quite happy with the current implementation tbh. It's flexible, allows passing anything as an icon and takes care of padding needed to space out the icon and the text.

Reply:
`<wof-button type="accent">
        <template v-slot:icon>
        <wof-icon class="wof-like" />
        </template>
        An accent button
 </wof-button>`
But using this is not very comfortable. 

`<wof-button type="accent" icon="wof-like">
        An accent button
 </wof-button>`
This is better in my opinion

Reply:
I don't quite agree, it introduces fragile coupling between components. Change anything in WofIcon (styles e.g.) and it implies fixing the button because it expected something which won't be true anymore.

Reply:
I updated it according your instructions but I am still not convinced :P

Reply:
It is very unlikely that we will change the implementation of the icon. It is so small that it is not worth preparing for it. Especially in button.
It would be true in bigger project.